### PR TITLE
chore(nodebuilder)!: Elevate perms for certain endpoints

### LINKED ISSUE #2406: nodebuilder/header: Elevate permissions of subscription endpoint

### PR DESCRIPTION
**Pull request by renaynay**
TODO: 
- [x] should perms be elevated for `share` endpoints like `GetEDS` ? IMO yes.

Resolves #2406

### CODE REVIEW

File: nodebuilder/state/state.go [hunk 1]
```diff
@@ -150,11 +150,11 @@ type API struct {
 		QueryDelegation func(
 			ctx context.Context,
 			valAddr state.ValAddress,
-		) (*types.QueryDelegationResponse, error) `perm:"public"`
+		) (*types.QueryDelegationResponse, error) `perm:"read"`
 		QueryUnbonding func(
 			ctx context.Context,
 			valAddr state.ValAddress,
-		) (*types.QueryUnbondingDelegationResponse, error) `perm:"public"`
+		) (*types.QueryUnbondingDelegationResponse, error) `perm:"read"`
```
Comment (Wondertan) [inline nodebuilder/state/state.go:157]: Why not public?

**Comment by renaynay** [comment]
@Wondertan actually we should at least elevate perms of SharesAvailable as there is a sanity check there that could cause node to panic in case of malformed root -- I also elevated perms of the other calls as i don't think calls that require node to do work should be public.

**Comment by distractedm1nd** [comment]
Read permissions are sufficient, even for SharesAvailable with the panic. Public methods are endpoints you can give anyone access to, read perms are next level up and should only be given to people who are not using with ill intent anyways

File: nodebuilder/state/state.go (hunk 1 above)
Comment (distractedm1nd) [reply nodebuilder/state/state.go:157]: True. We should maybe think about making these methods take the AccountAddress as well

**Review by distractedm1nd — APPROVED** [review]

**Comment by renaynay** [comment]
@distractedm1nd i made the comment re SharesAvailable before i elevated the perm to read

**Comment by distractedm1nd** [comment]
Please mark as breaking

**Comment by renaynay** [comment]
already implemented

**Pull request closed without merging**